runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpPauseState.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.xptracker;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.EnumMap;
+import java.util.Map;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.Skill;
+
+public class XpPauseState
+{
+	// Internal state
+	private final Map<Skill, XpPauseStateSingle> skillPauses = new EnumMap<>(Skill.class);
+	private boolean cachedIsLoggedIn = false;
+
+	/**
+	 * Tick the pause status for all skills, taking into account skill no-xp timeouts
+	 * and logout pause settings
+	 */
+	public void tick(Client client, XpTrackerConfig config)
+	{
+		tickXp(client, config);
+		tickLogout(config, client.getGameState());
+	}
+
+	public boolean pauseSkill(Skill skill)
+	{
+		return findPauseState(skill).onManualPause();
+	}
+
+	public boolean unpauseSkill(Skill skill)
+	{
+		return findPauseState(skill).onUnPause(Instant.now());
+	}
+
+	public boolean isPaused(Skill skill)
+	{
+		return findPauseState(skill).isPaused();
+	}
+
+	// Private internal functions
+	private void tickXp(Client client, XpTrackerConfig config)
+	{
+		final int pauseAfterMinutes = config.pauseSkillAfter();
+		final int pauseAfterSeconds = pauseAfterMinutes * 60;
+		final Instant now = Instant.now();
+		for (Skill skill : Skill.values())
+		{
+			final int currentXp = client.getSkillExperience(skill);
+			final XpPauseStateSingle state = findPauseState(skill);
+			if (state.getXp() != currentXp)
+			{
+				state.onXpChanged(currentXp, now);
+			}
+			else
+			{
+				final Instant lastChange = state.getLastChange();
+				// When config.pauseSkillAfter is 0, it is effectively disabled
+				if (pauseAfterMinutes > 0 && lastChange != null
+					&& Duration.between(lastChange, now).getSeconds() >= pauseAfterSeconds)
```

Comment:
Instead of creating duration all the time you can probably just compare milliseconds.